lib/ts/recipe/session/sessionClass.ts

```diff
@@ -61,15 +66,15 @@ export default class Session implements SessionContainerInterface {
                 this.reqResInfo.res,
                 this.reqResInfo.transferMethod,
                 this.reqResInfo.req,
-                userContext === undefined ? makeDefaultUserContextFromAPI(this.reqResInfo.req) : userContext
+                ctx
             );
         }
     }
 
-    async getSessionDataFromDatabase(userContext?: any): Promise<any> {
+    async getSessionDataFromDatabase(userContext?: Record<string, any>): Promise<any> {
         let sessionInfo = await this.helpers.getRecipeImpl().getSessionInformation({
             sessionHandle: this.sessionHandle,
-            userContext: userContext === undefined ? {} : userContext,
+            userContext: getUserContext(userContext),
```

Comment:
why not check for this.reqResInfo being undefined or not like it's done in the revokeSession function of this file?

Reply:
Applies to all functions in this file

Reply:
Fixed all functions